src/main/java/hudson/plugins/templateproject/ItemListenerImpl.java

```diff
@@ -47,11 +62,46 @@ public void onLoaded() {
 		}
 	}
 
+	@Override
+	public void onRenamed(Item item, String oldName, String newName) {
+		for (AbstractProject project : hudson.getAllItems(AbstractProject.class)) {
+			updateHandler.updateScm(project, oldName, newName);
+
+			boolean updatedItems = false;
+
+			if(project instanceof Project){
```

Comment:
I don't like this approach too, but how can I obtain project builders using only AbstractProject?